app.py

```diff
@@ -413,9 +440,6 @@ def prepare_generic_launch(self, engine_name, context, app_path, app_args):
         except Exception:
             self.log_exception("Error executing engine bootstrap script.")
             raise TankError("Error executing bootstrap script. Please see log for details.")
-        finally:
```

Comment:
I'm in two minds about removing this - fundamentally it's fine as the calling code now pushes/pops the environment but at the same time it's making the assumption that the calling code will __always__ push/pop the environment which feels a little bit wrong!